### ISSUE TITLE
about YOLOv11 coco augmentation

### ISSUE THREAD

**Issue by soyeollee** [ultralytics/ultralytics#16607]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello!

After checking the yolov11x.pt trained with coco, I found that the following settings were applied:

```
model.ckpt['train_args']['mosaic'] = 0.0  # yolov8x = 1.0
model.ckpt['train_args']['mixup'] = 0.0  # yolov8x = 0.15
model.ckpt['train_args']['copy_paste'] = 0.0  # yolov8x = 0.3
```
Q1) Is it correct that yolov11 was trained with these values?
Q2) Is there any specific reason for not using these augmentations?

Thank you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @soyeollee, thank you for reaching out to Ultralytics 🚀!

We appreciate you bringing your question about YOLOv11 coco augmentation to our attention. An Ultralytics engineer will review your inquiry soon.

In the meantime, here are some resources that might help:

- **Documentation:** Visit our [Docs](https://docs.ultralytics.com) for comprehensive guides, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.
- **Community Support:** Connect with us on [Discord](https://ultralytics.com/discord) 🎧 for real-time discussion, visit our [Discourse](https://community.ultralytics.com) for in-depth talks, or explore our [Subreddit](https://reddit.com/r/ultralytics).

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us address it efficiently.

For questions related to training or settings, sharing your training logs and configurations can be immensely helpful.

## Upgrade

Ensure you are using the latest `ultralytics` package. Update it along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Run YOLO in any of these up-to-date verified environments (with dependencies preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If green, all tests in Ultralytics' CI are passing. CI verifies the correct operation of YOLO modes and tasks daily and on each commit.

Thank you for your patience and understanding! 😊

**Comment by Y-T-G** [comment]
Yes. The `train_args` are what the model was trained with

**Comment by Laughing-q** [comment]
@Y-T-G @soyeollee Guys we were using augmentations during our yolo11 training, but somehow it seems the train_args was reset. I'll take a look in today and post all the augmentation args we were using here or #16602. 
Thanks for the reporting!

**Comment by Laughing-q** [comment]
@Y-T-G @soyeollee Yes I found the issue that some of the train_args would be overridden by close_mosaic, and I've fixed it by #16627.
Also the training arguments for yolo11 models are:
| model   | mosaic | mixup | copy_paste |
|---------|--------|-------|------------|
| yolo11n | 1.0    | 0.0   | 0.10       |
| yolo11s | 1.0    | 0.05  | 0.15       |
| yolo11m | 1.0    | 0.15  | 0.40       |
| yolo11l | 1.0    | 0.15  | 0.50       |
| yolo11x | 1.0    | 0.20  | 0.60       |

**Comment by ZZxiaoxiao** [comment]
I have a question regarding the use of the copy-paste method. Although YOLO has a new implementation, when I debug, I find that the segments in the labels are empty. To use this method, do I need to add additional mask data? I have experimented with adding and not adding the mask, but the results seem to be the same. Could you please teach me how to effectively use the copy-paste data augmentation method in YOLO for object detection tasks? Thank you so much.

**Comment by Laughing-q** [comment]
@ZZxiaoxiao yes you need to prepare segment labels to use copy_paste.

**Comment by ZZxiaoxiao** [comment]
Could you provide a label file that includes segmentation or a label file in YOLO format that can be used for training?
Should the data format in the .txt file be like this: class_id x_center y_center width height polygon_points?
use the txt of YOLO-seg??

**Comment by Laughing-q** [comment]
@ZZxiaoxiao Please prepare the dataset just like YOLO-seg models, here's a docs page you can refer to: https://docs.ultralytics.com/datasets/segment/

**Comment by ZZxiaoxiao** [comment]
If there is an issue with training parameters being overwritten, do the training parameters for YOLOv8s and YOLOv9s need to be updated? YOLOv8s only uses mosaic augmentation and does not use mixup or copy-paste data augmentation, correct? Has anyone reproduced the performance of YOLOv8s on the COCO dataset, and is it consistent with the official results? What about YOLOv9 and YOLOv10?

**Comment by ZZxiaoxiao** [comment]
thank you

**Comment by glenn-jocher** [comment]
@ZZxiaoxiao you're welcome! If you have any further questions, feel free to ask.

**Comment by IUnspace** [comment]
> [@Y-T-G](https://github.com/Y-T-G) @soyeollee Yes I found the issue that some of the train_args would be overridden by close_mosaic, and I've fixed it by [#16627](https://github.com/ultralytics/ultralytics/pull/16627). Also the training arguments for yolo11 models are:
> model 	mosaic 	mixup 	copy_paste
> yolo11n 	1.0 	0.0 	0.10
> yolo11s 	1.0 	0.05 	0.15
> yolo11m 	1.0 	0.15 	0.40
> yolo11l 	1.0 	0.15 	0.50
> yolo11x 	1.0 	0.20 	0.60


I just ducked into the default.yaml file https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/default.yaml

Noting that these don't match (only one default for all models) and copy-paste is for segmentation only anyway, is your above statement still true?  What am I missing here?

**Comment by Laughing-q** [comment]
@IUnspace yes still true, and we support using segment labels to train a detection model hence copy-paste could be applied to detection training this way.
